core/src/test/scala/org/broadinstitute/monster/ingester/core/IngestControllerSpec.scala

```diff
@@ -0,0 +1,368 @@
+
+  def withController(
+    jadeClient: JadeApiClient
+  )(test: (Transactor[IO], IngestController) => IO[Any]): Unit = {
+    val tx = transactor
+    test(tx, new IngestController(tx, jadeClient)).unsafeRunSync()
+    ()
+  }
+
+  def withRequest(
+    jadeClient: JadeApiClient
+  )(test: (Transactor[IO], IngestController) => IO[Any]): Unit =
+    withController(jadeClient) { (tx, controller) =>
+      val setup = for {
+        _ <- List(request1Id, request2Id, request3Id).zipWithIndex.traverse_ {
+          case (id, i) =>
+            val ts = timestampSql(Instant.ofEpochMilli(nowMillis + i))
+            List(
+              fr"INSERT INTO requests (id, submitted, dataset_id) VALUES ($id, ",
+              Fragment.const(ts),
+              fr", $dataset1Id)"
+            ).combineAll.update.run.void
+        }
+        _ <- request1Jobs.traverse_ {
+          case (path, table) =>
+            sql"""INSERT INTO jobs
+                  (request_id, status, path, table_name)
+                  VALUES
+                  ($request1Id, ${JobStatus.Pending: JobStatus}, $path, $table)""".update.run.void
+        }
+        _ <- request2Jobs.traverse_ {
+          case (path, table, jadeId) =>
+            sql"""INSERT INTO jobs
+                  (request_id, jade_id, status, path, table_name)
+                  VALUES
+                  ($request2Id, $jadeId, ${JobStatus.Running: JobStatus}, $path, $table)""".update.run.void
+        }
+      } yield ()
+
+      setup.transact(tx).flatMap(_ => test(tx, controller))
+    }
+
+  private val apiEmpty = buildApi { _ =>
+    Resource.pure(Response[IO]())
+  }
+
+  private val apiIngest = buildApi { _ =>
+    val thetimestring =
+      OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+    Resource.liftF(
+      IO.pure(
+        Response[IO](
+          status = Status.Ok,
+          body = Stream.emits(
+            s"""{ "status_code": 200, "id": "${UUID
+              .randomUUID()}", "job_status": "running", "submitted": "$thetimestring" }""".getBytes
+          )
+        )
+      )
+    )
+  }
+
+  private val apiJobStatus = buildApi { req =>
+    val jobId = req.uri.toString().split("/").last
+    val thetimestring =
+      OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+    Resource.liftF(
+      IO.pure(
+        Response[IO](
+          status = Status.Ok,
+          body = Stream.emits(
+            s"""{ "status_code": 200, "id": "$jobId", "job_status": "succeeded", "completed": "$thetimestring" }""".getBytes
+          )
+        )
+      )
+    )
+  }
+
+  def getNow: IO[Instant] =
+    clk.realTime(scala.concurrent.duration.MILLISECONDS).map(Instant.ofEpochMilli)
+
+  behavior of "IngestController"
+
+  // createRequest
+  it should "return a UUID for valid inputs" in withController(apiEmpty) {
+    (_, controller) =>
+      for {
+        now <- getNow
+        a <- controller.createRequest(dataset1Id, now)
+      } yield {
+        a
```

Comment:
Should this be asserting something?

Reply:
Nope I'll remove it; that was my "is this thing even working" test which is captured in the test after it while asserting things